Sources/MapboxMaps/Foundation/DisplayLinkProtocol.swift

```diff
@@ -14,4 +14,30 @@ internal protocol DisplayLinkProtocol: AnyObject {
     func invalidate()
 }
 
+internal final class EmptyDisplayLink: DisplayLinkProtocol {
+    var timestamp: CFTimeInterval { .nan }
+    var duration: CFTimeInterval { .nan }
+    var preferredFramesPerSecond: Int = 0
+#if swift(>=5.5)
+    // Stored properties cannot be annotated with @available, so we
+    // store the value as an `Any` in `_untypedPreferredFrameRateRange` below
+    // and make this a computed property.
+    @available(iOS 15.0, *)
+    var preferredFrameRateRange: CAFrameRateRange {
+        get { return (_untypedPreferredFrameRateRange as? CAFrameRateRange) ?? .default }
+        set { _untypedPreferredFrameRateRange = newValue }
+    }
+
+    private var _untypedPreferredFrameRateRange: Any?
+#endif
```

Comment:
There's some non-trivial logic here. Please add unit tests to confirm default values and preferredFrameRate getter and setter.

Reply:
Indeed, added `EmptyDisplayLinkTests` to test this.